### PR TITLE
correct `attention_head_dim` for `JointTransformerBlock`

### PR DESCRIPTION
**Pull request by yiyixuxu**

### CODE REVIEW

**Comment by yiyixuxu** [comment]
should finish https://github.com/huggingface/diffusers/pull/6893 and https://github.com/huggingface/diffusers/pull/7027

File: src/diffusers/models/controlnet_sd3.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def __init__(
                 JointTransformerBlock(
                     dim=self.inner_dim,
                     num_attention_heads=num_attention_heads,
-                    attention_head_dim=self.inner_dim,
+                    attention_head_dim=attention_head_dim,
```
Comment (DN6) [inline src/diffusers/models/controlnet_sd3.py:84]: Should this also be `self.config.attention_head_dim` to match `transformer_sd3.py`?

File: src/diffusers/models/attention.py [hunk 2]
```diff
@@ -128,9 +128,9 @@ def __init__(self, dim, num_attention_heads, attention_head_dim, context_pre_onl
             query_dim=dim,
             cross_attention_dim=None,
             added_kv_proj_dim=dim,
-            dim_head=attention_head_dim // num_attention_heads,
```
Comment (DN6) [inline src/diffusers/models/attention.py:131]: This won't break? Wouldn't the value of `dim_head` be computed differently?

File: src/diffusers/models/controlnet_sd3.py (hunk 1 above)
Comment (yiyixuxu) [inline src/diffusers/models/controlnet_sd3.py:84]:
```suggestion
                    attention_head_dim=self.config.attention_head_dim,
```

File: src/diffusers/models/attention.py (hunk 2 above)
Comment (DN6) [reply src/diffusers/models/attention.py:131]: Ahh I see. Thanks for the explaining! 🙏🏽

**Review by DN6 — APPROVED** [review]

**Pull request merged**